### DISCUSSION TITLE
AirflowFailException with PythonVirtualenvOperator

### DISCUSSION THREAD

**Discussion by rodrigomocca** [apache/airflow#16661] [General]
Hi!
I'm new with this tool and I really like it.
I have a doubt with the PythonVirtualenvOperator.
I'm using it to get _requests_ library, and I want to validate the response, if doen't fit the desired value, force the task to fail.
I was able to make the task fail raising a AirflowFailException, but this only works with the PythonOperator.
How can I make this work with the PythonVirtualenvOperator?

Thanks!

**Comment by josh-fell** [comment]
Can you explain more about what you mean by "make it work with the PythonVirtualenvOperator"?  Are you seeing an unexpected error, is the `AirflowFailException` not being thrown as it does using the `PythonOperator`, or something else?

**Reply by rodrigomocca** [reply]
Hi, I'm sorry, I've missed the response.
If I use the PythonVirtualenvOperator, the AirflowFailException works as any other exception, triggering the retries on the node.
If I use the PythonOperator and reise an AirflowFailException, no retries will be triggered, as expected.

I hope you can understand what I'm trying to explain, if not, please let me know and I'll try to explain my self a little bit better.

Thanks in advance

**Reply by kobethuwis** [reply]
Are your sure AirflowFailException is resolved correctly in your python-virtual-env? Please keep in mind that you need to specify all packages & modules yourself in the operator.
